### DISCUSSION TITLE
Explanation of pybamm.source(left, right, boundary=False)

### DISCUSSION THREAD

**Discussion by PaulKranefeld** [pybamm-team/PyBaMM#2554] [Q&A] [answered]
Hello, 
can someone explain me how this function works? `pybamm.source(left, right, boundary=False)` When do I use it? I have already read the documentation, but it is not clear to me why this function is necessary. Can I interpret this function as a source term within my governing equations? Do I have to use this function as soon as I have a source term? What happens if I do not use it? As you can see, the concept behind this function is unfortunately very unclear to me.
Thanks in advance for any kind of help.

**Comment by rtimms** [comment, marked as answer]
It's required for wrapping source terms when using the Finite Element Method. Currently, we only use FEM in 2D current collectors problems; everything else uses finite volumes. See [this issue](https://github.com/pybamm-team/PyBaMM/issues/1410) for a bit of a discussion. Ideally we'd automate this so the user doesn't need to worry about explicitly using `source`.

So, in summary:
- yes, this is a source term within the governing equations
- you don't have to use it unless you plan on using FEM (only used for 2D problems)
- if you don't use it there won't be any issues unless you use FEM

**Reply by PaulKranefeld** [reply]
Thanks for the quick feedback, that answers my question very well.
